### PR TITLE
(fix) Historical value extractor should support FHIR obs

### PR DESCRIPTION
**Pull request by samuelmale**
## Requirements

- [X] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

The obs value extractor currently supports Obs in the OpenMRS data modeling; This PR adds support for FHIR obs.

## Related Issue

https://openmrs.atlassian.net/browse/O3-4220

### CODE REVIEW

File: src/utils/form-helper.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import { type LayoutType } from '@openmrs/esm-framework';
-import { type OpenmrsObs, type FormField, type FormPage, type FormSection, type SessionMode } from '../types';
+import {
+  type FormField,
+  type FormPage,
+  type FormSection,
+  type SessionMode,
+  type FHIRObsResource,
+  type RenderType,
+} from '../types';
```
Comment (usamaidrsk) [inline src/utils/form-helper.ts:9]: nit

**Review by usamaidrsk — APPROVED** [review]
LGTM!

**Comment by pirupius** [comment]
@samuelmale am testing this out against some forms, do we need to leave the ending as `.then(obs => obs)` as per the other wrapper or do we need to specify the value?

**Review by pirupius — APPROVED** [review]
Thanks @samuelmale I will continue testing and communicate incase we find any edge cases.

**Pull request merged**